source/dev_guide/project-standards.rst

```diff
@@ -0,0 +1,154 @@
+Project Standards
+=================
+
+Development Environment
+-----------------------
+If you haven't already, please visit :ref:`"(b) Development Environment" <dev-env>`
+to set up your local development environment.
+
+Version Control
+---------------
+
+The repository uses a fork-based Git workflow with tag releases.
+
+.. figure:: git-flow.svg
+   :alt: Git Flow Diagram
+
+Guidelines
+~~~~~~~~~~
+1. ``master`` must always be **deployable**
+2. All changes are made through **support** branches on forks
+3. **Rebase** with ``master`` to avoid/resolve conflicts
+4. Make sure ``pre-commit`` checks pass when committing (enforced in CI/CD build)
+5. Open a pull-request (PR) early for discussion
+6. Once the CI/CD build passes and PR is approved, **squash and rebase** your
+   commits
+7. Merge PR into ``master`` and **delete the branch**
+
+Things to Avoid
+^^^^^^^^^^^^^^^
+-  Merge in broken or commented out code
+-  Committing onto ``master`` directly
+-  Merge with conflicts (handle conflicts upon rebasing)
+
+Source: https://gist.github.com/jbenet/ee6c9ac48068889b0912
+
+Pre-commit
+~~~~~~~~~~
+The repository uses the ``pre-commit`` package to manage pre-commit hooks.
+These hooks help enforce QA standards and identify simple issues
+at the commit level before submitting code reviews.
+
+.. figure:: pre-commit-flow.svg
+   :alt: Pre-commit Flow Diagram
+
+   ``pre-commit`` Flow
+
+Helpful Commands
+^^^^^^^^^^^^^^^^
+
+Install into your conda environment
+   ::
+
+        conda activate e3sm_diags_env_dev
+        pre-commit install
+
+Automatically run all pre-commit hooks (just commit)
+    ::
+
+        git commit -m '...'
+
+    .. figure:: ../pre-commit-passing.png
+       :alt: pre-commit Output
+
+       ``pre-commit`` Output
+
+Manually run all pre-commit hooks
+   ::
+
+        pre-commit run --all-files.
+
+Run individual hook
+   ::
+
+        # Available hook ids: trailing-whitespace, end-of-file-fixer, check-yaml, black, isort, flake8, mypy
+        pre-commit run <hook_id>.
+
+Squash and Rebase Commits
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before you merge a support branch back into ``master``, the branch is typically
+squashed down to a single buildable commit, and then rebased on top of the upstream ``master`` branch.
+
+-  Ensures build passes from the commit
+-  Cleans up Git history for easy navigation
+-  Makes collaboration and review process more efficient
+-  Makes handling conflicts from rebasing simple since you only have to
+   deal with conflicted commits
```

Comment:
I would preface this list with something like "why squash commits?"

Perhaps it would also be good to add the caveat that if the PR has isolated commits then they can stay separated (although I suppose that point is made moot by the idea that a PR should only be handling one specific thing).

Another benefit is that it makes `git bisect` easier. (e.g., if you break the code in commit 1 and fix it in commit 2, then `git bisect` might get caught on commit 1 rather than the actual faulty commit).

Reply:
Sounds good, as long as the isolated commits pass the build.

I'll also add `git bisect` to the list.